src/Transport/CliSnmpClient.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\PhpSnmp\Transport;
+
+use SimPod\PhpSnmp\Exception\EndOfMibReached;
+use SimPod\PhpSnmp\Exception\GeneralException;
+use SimPod\PhpSnmp\Exception\InvalidVersionProvided;
+use SimPod\PhpSnmp\Exception\NoSuchInstanceExists;
+use SimPod\PhpSnmp\Exception\NoSuchObjectExists;
+use SimPod\PhpSnmp\Exception\SnmpException;
+use SimPod\PhpSnmp\Transport\Cli\ProcessExecutor;
+use SimPod\PhpSnmp\Transport\Cli\SymfonyProcessProcessExecutor;
+use Throwable;
+use function array_merge;
+use function assert;
+use function explode;
+use function in_array;
+use function Safe\preg_match;
+use function Safe\substr;
+use function strlen;
+use function strpos;
+
+final class CliSnmpClient implements SnmpClient
+{
+    use SimpleBatch;
```

Comment:
asi bych to udelal pres extend nez pres traitu

Reply:
proč?

Reply:
pac traity jsou hacky a spis jsou s nima problemy. viz napr. jakejkoliv novej check ze strany static analysis se na nich rozbije :D 

prekvapuje me, ze tady vsechno prochazi, protoze traita vola metody, o kterejch nevi

Reply:
nevím proč by měli být hacky :D používat tu dědičnost je imo špatně, ty dvě třídy spolu nemají nic společnýho, jen tenhle shodný kus kódu

no, asi se ten tooling posunul - napovídá phpstorm, phpstan s tím nemá problém, ... it works.

Reply:
v budoucnu se to bude sem tam rozbijet no. w.e, ale ja osobne bych se tomu uplne vyhnul

Reply:
hele když se to rozbije tak budeme řešit, teď to issue není a mě osobně se to takhle líbí víc. Přijde mi takhle ok, trait je regulérní feature, to že s tím tooling neumí vždy dobře pracovat je jiná věc...

Reply:
trait v src vetsinou indikuje code smell no :D me kdyz se chce napsat traitu tak jdu fixnout design

Reply:
většinou souhlas, ale tady to je imo regulérně ok usecase ... mít base třídu by naopak byl bad design imo. Zavádět nějakou dědičnost jen protože mají část společnýho kódu imo není dobře.

Reply:
oboji je shit no, ale nechtelo se mi premejslet, jak to napsat jinak